seacrowd/sea_datasets/filipino_hatespeech_tiktok/filipino_hatespeech_tiktok.py

```diff
@@ -0,0 +1,131 @@
+from seacrowd.utils.constants import Tasks, Licenses
+
+_CITATION = """\
+@inproceedings{hernandez2021bert,
+  title={A bert-based hate speech classifier from transcribed online short-form videos},
+  author={Hernandez Urbano Jr, Rommel and Uy Ajero, Jeffrey and Legaspi Angeles, Angelic and Hacar Quintos, Maria Nikki and Regalado Imperial, Joseph Marvin and Llabanes Rodriguez, Ramon},
+  booktitle={2021 5th International Conference on E-Society, E-Education and E-Technology},
+  year={2021}
+}
+"""
+
+_LOCAL = False
+_LANGUAGES = ["tgl"]
+_DATASETNAME = "filipino_hatespeech_tiktok"
+
+_DESCRIPTION = """\
+The dataset contains annotated hate speech from transcribed Tiktok videos, mostly in Taglish (codemixed Tagalog and Cebuano) collected via an unofficial Tiktok API. Most of the domain and context of the videos are related to politics and general elections in the Philippines. Labeling: 0 - no hate speech, 1 - recognized hate speech
+"""
+
+_HOMEPAGE = "https://github.com/imperialite/filipino-tiktok-hatespeech"
+_LICENSE = Licenses.UNKNOWN.value
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+_URLS = {
+    "raw": "https://raw.githubusercontent.com/imperialite/filipino-tiktok-hatespeech/main/data",
+}
+
+
+class FilipinoHateSpeechTikTok(datasets.GeneratorBasedBuilder):
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="filipino_hatespeech_tiktok_source",
+            version=SOURCE_VERSION,
+            description="Filipino TikTok Hatespeech source schema",
+            schema="source",
+            subset_id="filipino_hatespeech_tiktok",
+        ),
+        SEACrowdConfig(
+            name="filipino_hatespeech_tiktok_seacrowd_text",
+            version=SEACROWD_VERSION,
+            description="Filipino TikTok Hatespeech Seacrowd schema",
+            schema="seacrowd_text",
+            subset_id="filipino_hatespeech_tiktok",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "filipino_tiktok_hatespeech_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features({"text": datasets.Value("string"), "label": datasets.Value("string")})
+        elif self.config.schema == "seacrowd_text":
+            features = schemas.text.features()
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        base_dir = _URLS["raw"]
+        data_files = {
+            "train": base_dir + "/train.csv",
+            "test": base_dir + "/test.csv",
+            "validation": base_dir + "/valid.csv",
+        }
+
```

Comment:
Not required, but it may be better to be explicit in the download step here, so that the typing is correct. We can add a line with something like:

```python
train_filepath = Path(dl_manager.download_and_extract(_URLS["train"]))
```